cli/azd/pkg/cloud/cloud.go

```diff
@@ -0,0 +1,94 @@
+package cloud
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
+)
+
+const (
+	ConfigPath = "cloud"
+
+	AzurePublicName       = "AzureCloud"
+	AzureChinaCloudName   = "AzureChinaCloud"
+	AzureUSGovernmentName = "AzureUSGovernment"
+)
+
+type PortalUrlBase = string
+
+type Cloud struct {
+	Configuration cloud.Configuration
+
+	// The base URL for the cloud's portal (e.g. https://portal.azure.com for
+	// Azure public cloud).
+	PortalUrlBase string
+
+	// The suffix for the cloud's storage endpoints (e.g. core.windows.net for
+	// Azure public cloud). These are well known values and can be found at:
+	// https://<management-endpoint>/metadata/endpoints?api-version=2023-12-01
+	StorageEndpointSuffix string
```

Comment:
Is there a way to get this via azure SDK?

Reply:
Storage blob clients in the Azure SDK for Go all require a URL or, in the case of a connection string, an account name and an endpoint suffix (e.g. core.windows.net). In the case of the suffix, the default suffix for Azure Public is hardcoded... but there isn't a direct listing of endpoint suffixes available in the SDK.